File: board/subjects.py
from django.db import models

class Subject(models.Model):

    SUBJECT_SELECTABLES = [
        ("OTH", "Other"),
        ("MET", "Meta"),
        ("WKS", "Weekly Summary"),
        ("ENS", "English Standard"),
        ("ENA", "English Advanced"),
        ("ENX", "English Extension 1"),
        ("EN2", "English Extension 2"),
        ("MAS", "Maths Standard 2"),
        ("MAT", "Maths Advanced"),
        ("MA1", "Maths Extension 1"),
        ("MA2", "Maths Accelerated"),
        ("BIO", "Biology"),
        ("CHE", "Chemistry"),
        ("PHY", "Physics"),
        ("INS", "Investigating Science"),
        ("ENG", "Engineering Studies"),
        ("SIC", "Studies in Catholic Thought"),
        ("SOR", "Studies of Religion I"),
        ("SRX", "Studies of Religion II"),
        ("ECO", "Economics"),
        ("BUS", "Business Studies"),
        ("LST", "Legal Studies"),
        ("HIA", "Ancient History"),
        ("HIM", "Modern History"),
        ("GEO", "Geography"),
        ("PDH", "PDHPE"),
        ("DRA", "Drama"),
        ("VAR", "Visual Arts"),
        ("MU1", "Music 1"),
        ("MU2", "Music 2"),
        ("FRE", "French Continuers"),
        ("LAT", "Latin Continuers"),
        ("JAP", "Japanese Continuers")]
    

# ... and the rest of your NESA HSC courses which are
# available from https://www.educationstandards.nsw.edu.au/wps/portal/nesa/11-12/stage-6-learning-areas/

# PotatoBoard dictates a Field.choices as a list iterable consisting of exactly two items
#  
#   [('A', 'B'), ('P', 'Q'), ('X', 'Y')]
#
# see https://docs.djangoproject.com/en/1.8/ref/models/fields/#choices for more info

File: board/static/board/subjects.js
var SUBJECT_ROLE_IDS = {
    "OTH": "NULL",
    "MET": "NULL",
    "WKS": "700120714811801631",
    "ENS": "807811981238861844",
    "ENA": "807811751750926356",
    "EN1": "807812013870415912",
    "EN2": "NULL",
    "MAS": "807812111190327306",
    "MAT": "807812159693783061",
    "MA1": "807812213955362816",
    "MA2": "807812213975810089",
    "BIO": "807812213980790784",
    "CHE": "807812222587502604",
    "PHY": "807812778424664095",
    "INS": "807812623634661387",
    "ENG": "807813063990181919",
    "SIC": "807812876916097024",
    "SOR": "807812817059053568",
    "SRX": "807812849644732426",
    "ECO": "807812429911687178",
    "BUS": "807812219693694976",
    "LST": "807812704453525504",
    "HIA": "807812512115327007",
    "HIM": "807812603740160010",
    "GEO": "807812512032096266",
    "PDH": "807812758061318225",
    "DRA": "807812222943887410",
    "VAR": "807812797332717578",
    "MU1": "807812726704439309",
    "MU2": "807812741946540093",
    "FRE": "807812452079108096",
    "LAT": "807812675413082173",
    "JAP": "807812651211816990"
}

/*

Making Discord role pings work with a key:pair dictionary that corresponds to role IDs. Role IDs are stored as strings.

PotatoBoard dictates a Field.choices as a list iterable consisting of exactly two items
  
    {A: "B", P: "Q", X: "Y"}

see https://docs.djangoproject.com/en/1.8/ref/models/fields/#choices for more info.

*/